ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +242,21 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0 && field->decimals < 7) {
+        length = mnd_sprintf(
+            &value,
+            0,
+            "%s%02u:%02u:%02u.%0*" ZEND_ULONG_FMT_SPEC,
+            (t.neg ? "-" : ""),
+            t.hour,
+            t.minute,
+            t.second,
+            field->decimals,
+           (int) (t.second_part / pow(10, 6 - field->decimals))
```

Comment:
This doesn't seem right, what was wrong with just left padding it up to the number of decimals?

Reply:
If the passed field width is smaller than the number of digits, all the digits would be included.

Reply:
Seems it always delivers 6 digits, the irrelevant places are filled with zeroes. Like with TIMESTAMP(3), a number would be like 123000. At least those DB servers I was able to test do that.

Reply:
That is exactly the case.